independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationsTransformer.java

```diff
@@ -64,4 +84,211 @@ default boolean isMethod() {
+            return containsAny(
+                    Arrays.stream(annotationNames).map(a -> DotName.createSimple(a.getName())).collect(Collectors.toList()));
+        }
+
+        /**
+         * {@link TransformationContext#getAnnotations()} must NOT contain any of the given annotations.
+         * 
+         * @param annotationNames
+         * @return self
+         */
+        public Builder containsNone(List<DotName> annotationNames) {
+            return and(context -> !Annotations.containsAny(context.getAnnotations(), annotationNames));
+        }
+
+        /**
+         * {@link TransformationContext#getAnnotations()} must NOT contain any of the given annotations.
+         * 
+         * @param annotationNames
+         * @return self
+         */
+        public Builder containsNone(DotName... annotationNames) {
+            return containsNone(List.of(annotationNames));
+        }
+
+        /**
+         * {@link TransformationContext#getAnnotations()} must NOT contain any of the given annotations.
+         * 
+         * @param annotationNames
+         * @return self
+         */
+        @SafeVarargs
+        public final Builder containsNone(Class<? extends Annotation>... annotationNames) {
+            return containsNone(
+                    Arrays.stream(annotationNames).map(a -> DotName.createSimple(a.getName())).collect(Collectors.toList()));
+        }
+
+        /**
+         * 
+         * @param transform
+         * @return self
+         */
+        public Builder transform(Consumer<TransformationContext> transform) {
+            this.transform = transform;
+            return this;
+        }
+
+        /**
+         * 
+         * @return a new annotation transformer
+         */
+        public AnnotationsTransformer build() {
+            Predicate<Kind> appliesTo = this.appliesTo;
+            int priority = this.priority;
+            Consumer<TransformationContext> transform = Objects.requireNonNull(this.transform);
+            Predicate<TransformationContext> predicate = this.predicate;
+            return new AnnotationsTransformer() {
+
+                @Override
+                public int getPriority() {
+                    return priority;
+                }
+
+                @Override
+                public boolean appliesTo(Kind kind) {
+                    return appliesTo != null ? appliesTo.test(kind) : true;
+                }
+
+                @Override
+                public void transform(TransformationContext context) {
+                    if (predicate == null) {
+                        transform.accept(context);
+                    } else if (predicate.test(context)) {
+                        transform.accept(context);
+                    }
+                }
+
+            };
+        }
+
+        /**
+         * 
+         * @param other
+         * @return self
+         */
+        public Builder and(Predicate<TransformationContext> other) {
```

Comment:
Maybe should be `private`? Or if `public`, then perhaps with a better name than `and`? (`predicate` comes to mind)

Reply:
Well, I don't like the `predicate` name but I'd like to keep it public so that users can add their "conditions" easily. How about `addCondition()` or something like that?

Reply:
Hm, `addCondition()` is not good either... `andTransformIf()`? Also not good :-/

Reply:
Yea I think the problem is that there are two kinds of conditions :-) You could rename `and` to `when` if you renamed `contains{All,Any,None}` to `whenContains{All,Any,None}`, for example. But not a big deal if not, I'd say.

Reply:
Hm, I like the `when()` idea...